static/app/views/dashboards/widgetBuilder/components/widgetOnDemandQueryWarning.tsx

```diff
@@ -0,0 +1,73 @@
+import type {Location} from 'history';
+
+import {OnDemandWarningIcon} from 'sentry/components/alerts/onDemandMetricAlert';
+import {tct} from 'sentry/locale';
+import type {PageFilters} from 'sentry/types/core';
+import {isOnDemandQueryString} from 'sentry/utils/onDemandMetrics';
+import {hasOnDemandMetricWidgetFeature} from 'sentry/utils/onDemandMetrics/features';
+import type {UseApiQueryResult} from 'sentry/utils/queryClient';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import useOrganization from 'sentry/utils/useOrganization';
+import {
+  OnDemandExtractionState,
+  type DashboardFilters,
+  type ValidateWidgetResponse,
+  type WidgetQuery,
+  type WidgetType,
+} from 'sentry/views/dashboards/types';
+
+interface Props {
+  canAddSearchConditions: boolean;
+  hideLegendAlias: boolean;
+  location: Location;
+  onAddSearchConditions: () => void;
+  onQueryChange: (queryIndex: number, newQuery: WidgetQuery) => void;
+  onQueryConditionChange: (isQueryConditionValid: boolean) => void;
+  onQueryRemove: (queryIndex: number) => void;
+  queries: WidgetQuery[];
+  selection: PageFilters;
+  validatedWidgetResponse: UseApiQueryResult<ValidateWidgetResponse, RequestError>;
+  widgetType: WidgetType;
+  dashboardFilters?: DashboardFilters;
+  projectIds?: number[] | readonly number[];
+  queryErrors?: Array<Record<string, any>>;
+}
```

Comment:
Other than `validatedWidgetResponse` field, could this interface also be removed?